dask_saturn/settings.py

```diff
@@ -4,49 +4,58 @@
 
 import os
 
-from typing import Optional
 from urllib.parse import urlparse
+from ._version import get_versions
+
+__version__ = get_versions()["version"]
 
 
 class Settings:
     """Global settings"""
 
     SATURN_TOKEN: str
-    BASE_URL: str
-
-    def __init__(
-        self,
-        base_url: Optional[str] = None,
-        saturn_token: Optional[str] = None,
-    ):
-        if base_url:
-            self.BASE_URL = base_url
-        else:
-            try:
-                self.BASE_URL = os.environ["BASE_URL"]
-            except KeyError as err:
-                err_msg = "Missing required value Saturn url."
+    SATURN_BASE_URL: str
+
+    def __init__(self):
+        try:
+            self.SATURN_BASE_URL = os.environ["SATURN_BASE_URL"]
+        except KeyError as err:
+            if os.environ.get("BASE_URL") is not None:
+                # if ``BASE_URL`` is set and ``SATURN_BASE_URL`` isn't, it's an old
+                # version of Saturn that is incompatible with this version of dask_saturn.
+                err_msg = (
+                    "This version of dask-saturn is incompatible with your Saturn version. "
+                    "Downgrade dask-saturn to `0.2.3`: `pip install dask_saturn==0.2.3`"
+                )
```

Comment:
If this PR is merged as-is and a new `dask-saturn` is published to PyPI before Saturn is upgraded, then anyone with `pip install --upgrade dask-saturn`  in their start script will see their Jupyter / deployment / job/ Prefect Cloud flow code fail. I think this PR's changes could be made in a way that avoids that risk of disruption.

Will you consider the options below?

**avoid publishing `dask-saturn` until a compatible version of Saturn has been rolled out to customers**

That's how a very similar situation with `prefect-saturn` was handled and I think it went well (https://github.com/saturncloud/prefect-saturn/pull/40).

**make the next release of `dask-saturn` compatible with both sets of environment variables**

I think it could work to do a release of `dask-saturn` that works like this:

1. look for `SATURN_BASE_URL`
2. if that isn't there use `BASE_URL`
3. if neither are present, raise an error

Reply:
Yeah good point. I should have said that in the top-level description. This should definitely not be released until a compatible version of Saturn has been rolled out. I just merged the changes in Atlas that enable that, so the release should be Monday and the rollout a few weeks after that.

Reply:
Ah got it, ok!

If that's the plan, then I think the places in this PR using `DeprecationWarning` are a bit misleading and that those should just be changed to `RuntimeError` (like I alluded to in https://github.com/saturncloud/dask-saturn/pull/58#discussion_r645811564).

I also recommend `dask-saturn`'s next version getting a minor- version bump (0.2.3 --> 0.3.0 or 1.0.0) to indicate the breaking changes, and to protect people who've pinned to `dask-saturn<0.3.0`

Reply:
yeah sorry I guess I put all the rollout thinking in https://github.com/saturncloud/saturn/pull/2402

Reply:
ah I did not see that, thanks! That table is very helpful